Update event descriptions in contracts.md (#7851)
Particular/docs.particular.net: servicecontrol/contracts.md

> Events described in this section are published by ServiceControl starting with version 4.17.

ServiceControl will also publish events related to archiving and retrying messages:
- `FailedMessagesArchived`: Event emitted for failed messages that were archived, indicating they won’t be retried
- `FailedMessagesUnArchived`: Event emitted for failed messages that were un-archived (restored from the archive), making them eligible for retry or further action

> - `MessageFailureResolvedByRetry`: Event emitted by ServiceControl for each failed message that was resolved by retry action

Reviewer: ```suggestion
- `MessageFailureResolvedByRetry`: Event emitted by ServiceControl for each failed message that succeeded after retrying
```
